### PR TITLE
USHIFT-1064: lightweight, unprivileged container image "MicroShift Test"

### PR DESCRIPTION
**Pull request by chiragkyal**
**Which issue(s) this PR addresses**:

Closes [USHIFT-1064](https://issues.redhat.com/browse/USHIFT-1064)

### CODE REVIEW

File: e2e/Dockerfile.hello-microshift
Comment (chiragkyal) [inline e2e/Dockerfile.hello-microshift:1]: Let me know if this is the preferred location for this file

File: e2e/Dockerfile.hello-microshift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM busybox:1.36
```
Comment (pmtk) [inline e2e/Dockerfile.hello-microshift:1]: Can we pick an image from quay? Just have in mind we need both amd and arm

File: e2e/Dockerfile.hello-microshift
Comment (pmtk) [reply e2e/Dockerfile.hello-microshift:1]: Currently we store Dockerfiles in `packaging/images/openshift-ci/` so we might want to move it there until we pick a better place for it when making decisions about e2e framework

File: e2e/Dockerfile.hello-microshift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+ARG USERNAME=microshift
+ARG GROUPNAME=$USERNAME
+ARG USER_UID=1001
+ARG USER_GID=$USER_UID
```
Comment (pmtk) [inline e2e/Dockerfile.hello-microshift:6]: Sanity question: have you tested the image on MicroShift? Let's be sure there are no security errors when deploying Pod with this image

**Comment by fzdarsky** [comment]
We have a "Hello MicroShift" image on Quay already (https://quay.io/repository/microshift/hello-microshift), is that insufficient? Source is here: https://github.com/redhat-et/microshift-demos/tree/main/apps/hello-microshift

**Comment by pmtk** [comment]
> We have a "Hello MicroShift" image on Quay already (https://quay.io/repository/microshift/hello-microshift), is that insufficient? Source is here: https://github.com/redhat-et/microshift-demos/tree/main/apps/hello-microshift

I saw that image, but didn't knew what's inside or where it's coming from. We might move (copy) it here to have everything in one place (although for testing I'd stay with plain text "Hello MicroShift" to not clutter logs with html)

**Comment by dhellmann** [comment]
> > We have a "Hello MicroShift" image on Quay already (https://quay.io/repository/microshift/hello-microshift), is that insufficient? Source is here: https://github.com/redhat-et/microshift-demos/tree/main/apps/hello-microshift
> 
> I saw that image, but didn't knew what's inside or where it's coming from. We might move (copy) it here to have everything in one place (although for testing I'd stay with plain text "Hello MicroShift" to not clutter logs with html)

If we link that image to the new container file, it would be rebuilt and the new version published. I can help set that up.

**Comment by fzdarsky** [comment]
> I saw that image, but didn't knew what's inside or where it's coming from. We might move (copy) it here to have everything in one place (although for testing I'd stay with plain text "Hello MicroShift" to not clutter logs with html)

The current image is for demos and users to have a “Hello, World!”-style example app when getting started. Therefore, it is deliberately visual and with logo.
For testing that isn’t needed, of course, but I wanted to flag that overwriting this image somewhat breaks the former use case and may surprise a few people.

**Comment by dhellmann** [comment]
> > I saw that image, but didn't knew what's inside or where it's coming from. We might move (copy) it here to have everything in one place (although for testing I'd stay with plain text "Hello MicroShift" to not clutter logs with html)
> 
> The current image is for demos and users to have a “Hello, World!”-style example app when getting started. Therefore, it is deliberately visual and with logo. For testing that isn’t needed, of course, but I wanted to flag that overwriting this image somewhat breaks the former use case and may surprise a few people.

Good point. Let's plan to make a new image for use in testing.

File: e2e/Dockerfile.hello-microshift (hunk 1 above)
Comment (chiragkyal) [reply e2e/Dockerfile.hello-microshift]: Created a busybox mirror `quay.io/microshift/busybox:1.36` with [skopeo-copy](https://github.com/containers/skopeo/blob/main/docs/skopeo-copy.1.md#examples). 

Busybox supports both `amd` and `arm`

```
podman manifest inspect quay.io/microshift/busybox:1.36
```

```json
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        ...
        ...
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 528,
            "digest": "sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        ...
        ...
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 528,
            "digest": "sha256:ba7000206594c2d72c3ab550453004c0dc50961157e5ebd2fb8ea1890099d02d",
            "platform": {
                "architecture": "arm64",
                "os": "linux",
                "variant": "v8"
            }
        }
    ]
}

```

However, I am unsure whether Podman `Build trigger` supports muti-arch build or not. If not, may be we need to think of a CI job for this.

File: e2e/Dockerfile.hello-microshift (hunk 2 above)
Comment (chiragkyal) [reply e2e/Dockerfile.hello-microshift]: Checked in `x86_64` Miroshift, the Pod was running fine with out any errors, and was able to access it through Route.

**Comment by chiragkyal** [comment]
> Good point. Let's plan to make a new image for use in testing.

Renamed this to "Microshift Test" to keep it separate for testing.

**Pull request merged**